README.md

```diff
@@ -1,99 +1,79 @@
+The TinyKV course builds a key-value storage system with the Raft consensus algorithm. After completing this course, you will have the knowledge to implement a horizontally scalable, highly available, key-value storage service with distributed transaction support.
 
-The whole project is a skeleton code for a kv server and a scheduler server at the beginning, and you need to finish the core logic step by step:
+The projects of this course are inspired by [MIT 6.824](https://pdos.csail.mit.edu/6.824/) and [TiKV Project](https://github.com/tikv/tikv).
 
-- Project1: build a standalone key-value server
-- Project2: build a highly available key-value server with Raft
-- Project3: support multi Raft group and balance scheduling on top of Project2
-- Project4: support distributed transaction on top of Project3
+## Course Architecture
 
-**Important note: This course is still under development, and the documentation is incomplete.** Any feedback and contribution is greatly appreciated. Please see help wanted issues if you want to join in the development.
+The whole project is a skeleton code for a key-value server and a scheduler server at the beginning - you need to finish the core logic step by step:
 
-## Course
+* [Standalone KV](doc/project1-StandaloneKV.md)
+  * Implement a standalone storage engine.
+  * Implement raw key-value service handlers.
+* [Raft KV](doc/project2-RaftKV.md)
+  * Implement the basic Raft algorithm
+  * Build a fault-tolerant KV server on top of Raft
+  * Add the support of Raft log garbage collection and snapshot
+* [Multi-raft KV](doc/project3-MultiRaftKV.md)
+  * Implement membership change and leadership change to Raft algorithm.
+  * Implement conf change and region split on Raft store
+  * Implement a basic scheduler.
+* [Transaction](doc/project4-Transaction.md)
+  * Implement the multi-version concurrency control layer.
+  * Implement handlers of `KvGet`, `KvPrewrite`, and `KvCommit` requests.
+  * Implement handlers of `KvScan`, `KvCheckTxnStatus`, `KvBatchRollback`, and `KvResolveLock` requests.
 
-Here is a [reading list](doc/reading_list.md) for the knowledge of distributed storage system. Though not all of them are highly related with this course, they can help you construct the knowledge system in this field.
-
-Also, you’d better read the overview of TiKV and PD's design to get a general impression on what you will build:
-
-- TiKV
-  - <https://pingcap.com/blog-cn/tidb-internal-1/> (Chinese Version)
-  - <https://pingcap.com/blog/2017-07-11-tidbinternal1/> (English Version)
-- PD
-  - <https://pingcap.com/blog-cn/tidb-internal-3/> (Chinese Version)
-  - <https://pingcap.com/blog/2017-07-20-tidbinternal3/> (English Version)
```

Comment:
Should not be removed.

Reply:
Let me try to add the reading list back. We don't display this message at the beginning IMO.

Reply:
Updated.